app.js

```diff
@@ -0,0 +1,84 @@
+var path = require('path');
+var fs = require('fs');
+var express = require('express');
+var Router = require('react-router');
+
+var indexStaticWatcher = require('./lib/index-static-watcher').create();
+var PORT = process.env.PORT || 8008;
+var PRODUCTION = (process.env.NODE_ENV === 'production');
+var DIST_DIR = path.join(__dirname, 'dist');
+
+var indexStatic;
+var router;
+var app = express();
+
+var startProdApp = function() {
+  console.log([
+    'Production mode enabled. Note that "npm install" is assumed to',
+    'have recently been run with NODE_ENV="production". If this is not',
+    'the case, some or all static assets may be out of date.'
+  ].join('\n'));
+  indexStaticWatcher.build(function(err, newIndexStatic) {
+    if (err) {
+      throw err;
+    }
+
+    console.log('Built server-side bundle.');
```

Comment:
Er sorry, yeah, this situation is kinda confusing now because this PR doesn't actually *replace* our static site generator with a dynamic lightweight server--it *adds* a lightweight dynamic server as an optional way to deploy the site.

To clarify, there's only one server intended for **production** here--it's the one in this PR, which dynamically renders HTML. There is no other production server, because our current site is just a bunch of pre-generated static files in `dist/` which we upload to S3.

However, there are actually 2 different kinds of servers we can use for **development**:

1. The development static file server, which is currently run when we use `npm start` (or `gulp watch` if you use that directly). This is a fairly "dumb" server, though, in the sense that it just kinda emulates S3 or any other kind of static file server.
2. The dynamic lightweight server, which is in this PR, and can be run via `npm run app` (or `gulp app`). Once this server is truly ready for production, we can potentially get rid of the development static file server, as well as the static site generator, and replace them with this.

That said, though, **we do actually generate (and copy) some static files even for the dynamic lightweight server**: These include e.g.:

* LESS/CSS files, 
* Client-side JavaScript files generated by webpack,
* Images

The above will only ever change when we push a new release, which is why, even with the lightweight dynamic server, we just generate/copy them once in `dist/`, and serve that directory as static files, just like we do with the purely static site we already use. In fact, the only thing that's *not* statically served in the dynamic lightweight server is every HTML page on the site.